lib/assist/assist.go

```diff
@@ -119,6 +129,29 @@ func (a *Assist) GenerateSummary(ctx context.Context, message string) (string, e
 	return a.client.Summary(ctx, message)
 }
 
+// GenerateCommandSummary summarizes the output of a command executed on one or
+// many nodes. The conversation history is also sent into the prompt in order
+// to gather context and know what information is relevant in the command output.
+func (a *Assist) GenerateCommandSummary(ctx context.Context, messages []*assist.AssistantMessage, output map[string][]byte) (string, error) {
+	// Create system prompt
+	modelMessages := []openai.ChatCompletionMessage{
+		{Role: openai.ChatMessageRoleSystem, Content: model.PromptSummarizeCommand},
+	}
+
+	// Load context back into prompt
+	for _, message := range messages {
+		role := kindToRole(MessageType(message.Type))
+		if role != "" && role != openai.ChatMessageRoleSystem {
```

Comment:
Did you find any message with an empty role? Curious about the first part of this statement.

Reply:
`COMMAND_RESULT` has an empty role for example